tests/functional/backend/corpora/test_collection_access.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+
+import requests
+from jose import jwt
+
+from tests.functional.backend.corpora.common import BaseFunctionalTestCase
+
+
+class TestCollectionAccess(BaseFunctionalTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.supercurator_token = cls.get_auth_token(
+            "supercurator@example.com",
+            cls.config.test_auth0_user_account_password,
+            additional_claims=["write:collections"],
+        )
+        cls.supercurator_cookie = cls.make_cookie(cls.supercurator_token)
+        cls.nocollection_token = cls.get_auth_token(
+            "nocollection@example.com",
+            cls.config.test_auth0_user_account_password,
+            additional_claims=["write:collections"],
+        )
+        cls.nocollection_cookie = cls.make_cookie(cls.nocollection_token)
+        cls.curator_token = cls.get_auth_token(
+            "nocollection@example.com",
+            cls.config.test_auth0_user_account_password,
+            additional_claims=["write:collections"],
+        )
+        cls.curator_cookie = cls.make_cookie(cls.curator_token)
+
+    def test_collection_access(self):
+        """Test that only a super curator has access to all of the collections"""
+        # get collections for nocollection user
+        headers = {"Cookie": f"cxguser={self.nocollection_cookie}", "Content-Type": "application/json"}
+        res = requests.get(f"{self.api}/dp/v1/collections", headers=headers)
+        self.assertEqual(res.status_code, requests.codes.ok)
+        # length should be 0
+        collections = res.json()["collections"]
+        private_collections = [c for c in collections if c["visibility"] == "PRIVATE"]
+        self.assertEqual(len(private_collections), 0)
+
+        # get collection for supercurator user
+        headers = {"Cookie": f"cxguser={self.supercurator_cookie}", "Content-Type": "application/json"}
+        res = requests.get(f"{self.api}/dp/v1/collections", headers=headers)
+        self.assertEqual(res.status_code, requests.codes.ok)
+        # len should be a lot
+        superuser_collections = [c for c in res.json()["collections"] if c["visibility"] == "PRIVATE"]
+
+        # get collection for curator user
+        headers = {"Cookie": f"cxguser={self.curator_cookie}", "Content-Type": "application/json"}
+        res = requests.get(f"{self.api}/dp/v1/collections", headers=headers)
+        self.assertEqual(res.status_code, requests.codes.ok)
+
+        # len should be less than super curator
+        curator_collections = [c for c in res.json()["collections"] if c["visibility"] == "PRIVATE"]
+
+        self.assertLess(len(curator_collections), len(superuser_collections))
```

Comment:
So I'm not totally clear on the test account user strategy... are we stipulating that other accounts will maintain a number of private collections at all times so there will be some collections that the test user can't see? I _presume_ this will happen...? But are we relying on all our adhoc accounts to always have private collections in existence? 🤔